src/Metadata/Resource/Factory/ResourceMetadataFactoryTrait.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace ApiPlatform\Core\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use ApiPlatform\Core\Metadata\Resource\Operation;
+use ApiPlatform\Core\Metadata\Resource\PaginationMetadata;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use ApiPlatform\Core\Annotation\Resource;
+
+trait ResourceMetadataFactoryTrait
```

Comment:
I'm :-1: for this trait due to violation of SRP (single responsibility principle).

In any case, things are most likely to change once you're done rewriting for the current codebase.